packages/react-native/src/components/3DS/useThreeDSecure.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from "react";
+import { useRef } from "react";
+import { useEvervault } from "../EvervaultProvider";
+import { startSession, threeDSecureSession } from "./threeDSSession";
+import {
+  ThreeDSecureCallbacks,
+  UseThreeDSecureResponse,
+  UseThreeDSecureState,
+} from "./types";
+
+export const useThreeDSecure = (): UseThreeDSecureResponse => {
+  const { appUuid } = useEvervault();
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+  const [displayModal, setDisplayModal] = useState(false);
```

Comment:
Very nit picking but could we rename this var to `isOpen` or `isVisible` as it's now not strictly for the built in modal version.